### PR TITLE
Auto preset creation

### PR DESCRIPTION
**Pull request by Kingdorian**
This adds all the functionality for automatic preset creation.

### CODE REVIEW

**Comment by jochem725** [comment]
There are some PMD and CheckStyle errors on this branch.

File: backend/src/main/java/com/benine/backend/preset/autopresetcreation/AutoPresetCreator.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      throw new CameraBusyException("The camera is busy.", cam.getId());
+    }
+    cam.setBusy(true);
+    ArrayList<IPCameraPreset> presets = new ArrayList<>();
+
+    cam.setBusy(false);
+    cam.setAutoFocusOn(true);
+    cam.setBusy(true);
+    for (ZoomPosition pos : generatePositions(cam, subViews)) {
+      cam.setBusy(false);
```
Comment (jochem725) [inline backend/src/main/java/com/benine/backend/preset/autopresetcreation/AutoPresetCreator.java]: Maybe the code in this for loop can be replaced by a method, improving readability.

**Comment by jochem725** [comment]
Apart from a small comment, I think this can be merged.

**Comment by bvanwijk** [comment]
Looks good, also since we agreed on most of the changes in seperate branches.
I think you can merge it.

**Pull request merged**